### PR TITLE
Feature/grid enhencement master

### PR DESCRIPTION
**Pull request by bh86**
- Implement sorting and filters in the grid component.
- Fix several issues in dato-grid

### CODE REVIEW

File: lib/src/grid/grid-toolbar/grid-toolbar.ts [hunk 1]
```diff
@@ -64,5 +64,5 @@ export interface ToolbarAction {
   text?: string;
   icon?: string;
   showWhen?: RowSelectionType | showWhenFunc;
-  click?: () => void;
+  click?: (selectedRows: any[]) => void;
```
Comment (NetanelBasal) [inline lib/src/grid/grid-toolbar/grid-toolbar.ts]: Add generic instead any.

File: lib/src/grid/grid/grid.theme.scss [hunk 2]
```diff
@@ -157,6 +186,104 @@ $params: (
...
+      padding-bottom: 0;
+    }
+
+    select.ag-filter-select {
+      -webkit-appearance: none;
+      -moz-appearance: none;
+      appearance: none;
+
+      line-height: 22px;
+      background: url("data:image/svg+xml;utf8, <svg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 10 6'><path fill-rule='evenodd' d='M9.392.92a.513.513 0 0 0-.923-.348l-3.59 3.733L1.29.572a.515.515 0 0 0-.738.718l4 4.102a.513.513 0 0 0 .738 0l4-4.102a.513.513 0 0 0 .103-.37z'/></svg>") 96% / 6% no-repeat currentColor;
```
Comment (NetanelBasal) [inline lib/src/grid/grid/grid.theme.scss]: I have an issue with an SVG background for IE in `nbChoices`, I would check if IE supports this.

Comment (NetanelBasal) [reply lib/src/grid/grid/grid.theme.scss]: https://github.com/NetanelBasal/nb-choices/issues/6

File: playground/src/app/components-gallery/previews/grid-preview/actions/grid-actions-preview.component.ts [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        value: 'three'
+      }];
+  }
+
+  ready(grid) {
+    this.gridApi = grid.api;
+    this.setRows(this.getRows());
+  }
+
+  private deleteRow(selectedRows: any[]) {
```
Comment (NetanelBasal) [inline playground/src/app/components-gallery/previews/grid-preview/actions/grid-actions-preview.component.ts]: use DatoGridAPI

File: playground/src/app/components-gallery/previews/grid-preview/grid-preview.component.ts [hunk 4]
```diff
@@ -85,11 +94,16 @@ export class GridPreviewComponent implements OnInit {
...
     return timer(3000).pipe(mapTo(rows))
-
   }
 
   myCustomLogic( selectedRows : any[] ) {
     return selectedRows.some(( row ) => row.value === 5);
   }
 
+  private deleteRow(selectedRows : any[]) {
+    return this.gridApi.updateRowData({
```
Comment (NetanelBasal) [inline playground/src/app/components-gallery/previews/grid-preview/grid-preview.component.ts]: same here

File: lib/src/grid/grid/grid.theme.scss (hunk 2 above)
Comment (bh86) [reply lib/src/grid/grid/grid.theme.scss]: @NetanelBasal it works in FF

**Pull request merged**